projects/pgai/pgai/cli.py

```diff
@@ -378,14 +388,24 @@ async def async_run_vectorizer_worker(
                 err_msg = f"unable to connect to database: {str(e)}"
             else:
                 err_msg = f"unexpected error: {str(e)}"
-            await handle_error(err_msg, None, worker_tracking, exit_on_error)
+            await handle_error(err_msg, vectorizer_id, worker_tracking, exit_on_error)
+        except BaseExceptionGroup as e:  # type: ignore
+            # catch any exceptions, log them, and keep on going
+            for exception in e.exceptions:  # type: ignore
+                error_msg = str(exception)  # type: ignore
+                log.error(error_msg, vectorizer_id=vectorizer_id)
+            for exception_line in traceback.format_exception(e):  # type: ignore
+                for line in exception_line.rstrip().split("\n"):
+                    log.debug(line)
```

Comment:
@JamesGuthrie We support python 3.10 as the minimum version, the builtin version of `BaseExceptionGroup` was added in 3.11.

When using the builtin version, the traceback includes the stacktraces of all the grouped exceptions. The issue is only with version from 3.10, which just has the stacktrace of where BaseExceptionGroup was raised. In either case we do still need to log the errors.

We are using python3.12 in the Docker image, so I'd just leave it like this. Maybe even raise our python version to 3.11.